core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -182,7 +181,49 @@ public void run() {
     for (Thread t : threads) {
       t.join();
     }
-    Mockito.verify(fileSystemContext).reset();
+    Mockito.verify(mMockFileSystemContext).reset();
+  }
+
+  /**
+   * Tests that failing to connect to Alluxio master causes initialization failure.
+   */
+  @Test
+  public void initializeFailToConnect() throws Exception {
+    Mockito.doThrow(ConnectionFailedException.class)
+        .when(mMockFileSystemMasterClient).connect();
+
+    URI uri = URI.create(Constants.HEADER + "randomhost:400/");
+    try {
```

Comment:
any reason not using [ExpectedException](http://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html) provided by JUnit (supposed to be a recommended way for exception assertion)?